src/Aurora.Infrastructure/Scrapers/FootFetishBooruImageGifScraper.cs

```diff
@@ -49,60 +50,61 @@ public async Task<List<SearchItem>> ScrapAsync(string term, CancellationToken to
-                            }
-                            var items = new List<SearchItem>();
-                            for (int i = 0; i < lastPageIndex; i++)
-                            {
-                                items.AddRange(await LoadPageAsync(term, client, i));
-                                if (config.UseLimitations && items.Count > config.MaxItemsCount)
-                                {
-                                    break;
-                                }
-                            }
-                            result = items;
-                        }
-                        else
-                        {
-                            _logger.LogInformation("Failed to parse index of the last image for term '{term}' at '{url}'", term, fullUrl);
-                            result = await LoadPageAsync(term, client, 0);
-                        }
+                        lastPageIndex = Math.Min(pagesCount, config.MaxPagesCount);
                     }
-                    //single page
                     else
                     {
-                        result = await LoadPageAsync(term, client, 0);
+                        lastPageIndex = pagesCount;
+                    }
+                    var items = new List<SearchItem>();
+                    for (int i = 0; i < lastPageIndex; i++)
+                    {
+                        items.AddRange(await LoadPageAsync(term, client, i));
+                        if (config.UseLimitations && items.Count > config.MaxItemsCount)
+                        {
+                            break;
+                        }
                     }
+                    return items;
+                },
+                _ => Task.FromResult(_emptyResult));
+            }
+            else
+            {
+                result = _emptyResult;
+            }
+            return result;
+        }
+
+        internal static ValueOrNull<int> ExtractPagesCount(HtmlDocument searchPage)
+        {
+            ValueOrNull<int> result;
+            var paginator = searchPage.DocumentNode.SelectSingleNode("//div[@id='paginator']");
+            var lastButton = paginator.ChildNodes.Last();
+            if (lastButton is not null && lastButton.GetAttributeValue("alt", "none") == "last page")
+            {
+                const string defVal = "none";
+                var lastButtonReference = lastButton.GetAttributeValue("href", defVal);
+                var pidPart = lastButtonReference.Split("&amp;").Where(x => x.StartsWith("pid")).FirstOrDefault();
+                var lastPidStr = pidPart?.Split('=')?.LastOrDefault();
+                if (lastPidStr is not null && Int32.TryParse(lastPidStr, out int lastPid))
+                {
+                    result = lastPid / _itemsPerPage + 1;
                 }
                 else
                 {
-                    result = _emptyResult;
+                    result = 1;
```

Comment:
return 1

Reply:
what is 1

Reply:
why is 1

Reply:
It indicates the number of pages. If there are no pager, then there is only one page available